### PR TITLE
[SQL Query Receiver] Add integration test for MySQL

### PR DESCRIPTION
**Pull request by dmolenda-sumo**
**Description:** Add integration test for MySQL in SQL Query Receiver

### CODE REVIEW

File: receiver/sqlqueryreceiver/testdata/integration/Dockerfile.mysql
Comment (astencel-sumo) [inline receiver/sqlqueryreceiver/testdata/integration/Dockerfile.mysql:1]: We should probably pin to a specific version of the [image](https://hub.docker.com/_/mysql)? Like `mysql:8.0.32` or perhaps only pin the minor version with `mysql:8.0`?

Comment (dmolenda-sumo) [reply receiver/sqlqueryreceiver/testdata/integration/Dockerfile.mysql:1]: good idea

File: receiver/sqlqueryreceiver/testdata/integration/Dockerfile.mysql
Comment (astencel-sumo) [inline receiver/sqlqueryreceiver/testdata/integration/Dockerfile.mysql:8]: Nit: would be nice to have the newlines at the end of files.

**Review by andrzej-stencel — APPROVED** [review]
This is great. I believe the integration tests are very valuable, ideally this component should have them for every db engine it supports. This MySQL test is a great addition to the already existing Oracle and PostgreSQL integration tests.

Thanks a lot Dominika for this contribution! :clap:

File: receiver/sqlqueryreceiver/integration_test.go
Comment (jpkrohling) [inline receiver/sqlqueryreceiver/integration_test.go:247]: Are we _really_ looking to wait up to 2 minutes of CI time for the port to become available? Can we speed this up a bit? How long would we typically wait?

**Comment by jpkrohling** [comment]
I'm approving the tests, so that we know how long it would actually take on a real CI environment.

File: receiver/sqlqueryreceiver/integration_test.go
Comment (dmolenda-sumo) [reply receiver/sqlqueryreceiver/integration_test.go:247]: 2 minutes is the maximum time we will wait for the container to be available. Usually, it takes a little less time. On my machine, it took less than 1 minute.

The default value of this parameter is one minute. However, it is increased in all other integration tests for this module: to [2 minutes for PostgreSQL](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/sqlqueryreceiver/integration_test.go#L40) and to [15 minutes for Oracle](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/sqlqueryreceiver/integration_test.go#L173). I decided to set it to 2 minutes (like for PosgreSQL), because on my laptop the container for MySQL takes a little longer to start than the one for PostgreSQL.

Of course, after merging, I will observe whether the test is performed correctly and if there is a need to adjust this time, I will do so

Comment (jpkrohling) [reply receiver/sqlqueryreceiver/integration_test.go:247]: If this takes about a minute to be ready, we might want to split this into a different step/suite so that it doesn't block the entire test suite for a whole minute. Our CI is already too slow for my taste, and we can't afford to have components add slow operations.

Comment (dmolenda-sumo) [reply receiver/sqlqueryreceiver/integration_test.go:247]: This test is part of the integration test suite. In my opinion, this is the test suite where we should put the slower ones. 

Running the test for MySQL requires less than a minute on our CI. [It took 158.981s to run all 3 integration tests in the sqlqueryreceiver module](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4637478361/jobs/8244496982?pr=20744#step:5:226). [Whole integration tests run in just 16m 27s](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4637478361/jobs/8244496982?pr=20744). To me, it doesn't look like too much execution time for such a large project.

Just like @astencel-sumo , I believe that such a component should have integration tests for each supported database

**Review by jpkrohling — APPROVED** [review]
I'm approving even though I don't agree that adding one extra minute for a single test is scalable in the bigger picture.

**Comment by astencel-sumo** [comment]
@dmitryax or @pmcollins can you take a look? Probably would be nice to get an opinion from a code owner or two. :wink:

**Review by pmcollins — APPROVED** [review]
LGTM. Thank you @dmolenda-sumo.

**Comment by astencel-sumo** [comment]
This looks like it's ready to merge. Are we waiting for something specific before merging this? @jpkrohling

**Review by atoulme — APPROVED** [review]

**Pull request merged**
